File: Backend-Java/car-catalog-backend/src/main/java/com/wswork/carcatalog/entity/Marca.java
package com.wswork.carcatalog.entity;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "marcas")
public class Marca {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "nome_marca", nullable = false, unique = true)
    private String nomeMarca;
    
    @Column(name = "data_criacao")
    private LocalDateTime dataCriacao;
    
    @Column(name = "data_atualizacao")
    private LocalDateTime dataAtualizacao;
    
    // Construtores
    public Marca() {}
    
    public Marca(String nomeMarca) {
        this.nomeMarca = nomeMarca;
    }
    
    // Getters e Setters
    public Long getId() {
        return id;
    }
    
    public void setId(Long id) {
        this.id = id;
    }
    
    public String getNomeMarca() {
        return nomeMarca;
    }
    
    public void setNomeMarca(String nomeMarca) {
        this.nomeMarca = nomeMarca;
    }
    
    public LocalDateTime getDataCriacao() {
        return dataCriacao;
    }
    
    public void setDataCriacao(LocalDateTime dataCriacao) {
        this.dataCriacao = dataCriacao;
    }
    
    public LocalDateTime getDataAtualizacao() {
        return dataAtualizacao;
    }
    
    public void setDataAtualizacao(LocalDateTime dataAtualizacao) {
        this.dataAtualizacao = dataAtualizacao;
    }
    
    // Callbacks JPA
    @PrePersist
    protected void aoCriar() {
        dataCriacao = LocalDateTime.now();
        dataAtualizacao = LocalDateTime.now();
    }
    
    @PreUpdate
    protected void aoAtualizar() {
        dataAtualizacao = LocalDateTime.now();
    }
    
    @Override
    public String toString() {
        return "Marca{" +
                "id=" + id +
                ", nomeMarca='" + nomeMarca + '\'' +
                ", dataCriacao=" + dataCriacao +
                ", dataAtualizacao=" + dataAtualizacao +
                '}';
    }
}

File: backend-kotlin/car-catalog-kotlin/bin/main/com/wswork/carcatalog/entity/Marca.kt
package com.wswork.carcatalog.kotlin.entity

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "marcas")
data class Marca(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    
    @Column(name = "nome_marca", nullable = false, unique = true)
    val nomeMarca: String,
    
    @Column(name = "data_criacao")
    val dataCriacao: LocalDateTime = LocalDateTime.now(),
    
    @Column(name = "data_atualizacao")
    val dataAtualizacao: LocalDateTime = LocalDateTime.now(),
    
    @OneToMany(mappedBy = "marca", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    val modelos: MutableList<Modelo> = mutableListOf()
)
